src/megaapi_impl.cpp

```diff
@@ -9395,7 +9539,24 @@ void MegaApiImpl::authorizeMegaNodePrivate(MegaNodePrivate *node)
             MegaNodePrivate *privNode = (MegaNodePrivate *)children->get(i);
             authorizeMegaNodePrivate(privNode);
         }
-    }  
+    }
+}
+
+MegaNode *MegaApiImpl::authorizeChatNode(MegaNode *node, const char *cauth)
+{
+    if (!node)
+    {
+        return NULL;
+    }
+
+    sdkMutex.lock();
```

Comment:
I think that it's not needed to lock/unlock the mutex in this function because it doesn't access anything from the SDK

Reply:
Agree. Removing the lock.